core/src/test/java/org/elasticsearch/action/main/MainResponseTests.java

```diff
@@ -55,8 +56,10 @@ public void testFromXContent() throws IOException {
         XContentType xContentType = randomFrom(XContentType.values());
         boolean humanReadable = randomBoolean();
         BytesReference originalBytes = toShuffledXContent(mainResponse, xContentType, ToXContent.EMPTY_PARAMS, humanReadable);
+        // we add a few random fields to check that parser is lenient on new fields
+        BytesReference withRandomFields = insertRandomFields(xContentType, originalBytes, null, random());
```

Comment:
In this case I was going with only one test, the xContent structure we test here is not that deep. My thinking was to do this kind of tests for smaller xContent objects and the two separate tests for deeper, top level tests (like SearchHit) where randomization already adds a lot of variety. I will keep this as is for now but will keep an eye on how this goes as we use this in other tests.